modules/atomic-widgets/prop-types/size-prop-type.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Elementor\Modules\AtomicWidgets\PropTypes;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+class Size_Prop_Type extends Transformable_Prop_Type {
+	protected array $default_units = [ 'px', 'em', 'rem', '%' ];
+
+	public function __construct() {
+		$this->internal_types['size'] = Number_Prop_Type::make();
+		$this->internal_types['unit'] = String_Prop_Type::make()->enum( $this->default_units );
+	}
+
+	public static function get_key(): string {
+		return 'size';
+	}
+
+	public function units( array $units ): self {
+		$this->internal_types['unit'] = $this->internal_types['unit']->enum( $units );
```

Comment:
```suggestion
		$this->internal_types['unit']->enum( $units );
```

it should work, it is a change by reference